pkg/ingestor/parser/parser_test.go

```diff
@@ -15,210 +15,192 @@
-// 				return
-// 			}
-// 			for i := range got {
-// 				compare(t, got[i].Edges, tt.want[i].Edges, got[i].Nodes, tt.want[i].Nodes)
-// 			}
-// 		})
-// 	}
-// }
-//
-// func compare(t *testing.T, gotEdges, wantEdges []assembler.GuacEdge, gotNodes, wantNodes []assembler.GuacNode) {
-// 	if !testdata.GuacEdgeSliceEqual(gotEdges, wantEdges) {
-// 		t.Errorf("ParseDocumentTree() = %v, want %v", gotEdges, wantEdges)
-// 	}
-// 	if !testdata.GuacNodeSliceEqual(gotNodes, wantNodes) {
-// 		t.Errorf("ParseDocumentTree() = %v, want %v", gotNodes, wantNodes)
-// 	}
-// }
-//
-// TODO: Fix tests to check for logger messages instead of err text (https://github.com/guacsec/guac/issues/765)
-// func Test_ParserSubscribe(t *testing.T) {
-// 	ctx := logging.WithLogger(context.Background())
-// 	err := verifier.RegisterVerifier(mockverifier.NewMockSigstoreVerifier(), "sigstore")
-// 	if err != nil {
-// 		if !strings.Contains(err.Error(), "the verification provider is being overwritten") {
-// 			t.Errorf("unexpected error: %v", err)
-// 		}
-// 	}
-//
-// 	natsTest := nats_test.NewNatsTestServer()
-// 	url, err := natsTest.EnableJetStreamForTest()
-// 	if err != nil {
-// 		t.Fatal(err)
-// 	}
-// 	defer natsTest.Shutdown()
-//
-// 	testCases := []struct {
-// 		name       string
-// 		tree       processor.DocumentTree
-// 		want       []assembler.AssemblerInput
-// 		wantErr    bool
-// 		errMessage error
-// 	}{{
-// 		name:       "valid dsse",
-// 		tree:       processor.DocumentTree(&dsseDocTree),
-// 		want:       graphInput,
-// 		wantErr:    true,
-// 		errMessage: context.DeadlineExceeded,
-// 	}, {
-// 		name:       "valid big SPDX document",
-// 		tree:       processor.DocumentTree(&spdxDocTree),
-// 		want:       spdxGraphInput,
-// 		wantErr:    true,
-// 		errMessage: context.DeadlineExceeded,
-// 	}}
-// 	for _, tt := range testCases {
-// 		t.Run(tt.name, func(t *testing.T) {
-// 			jetStream := emitter.NewJetStream(url, "", "")
-// 			ctx, err = jetStream.JetStreamInit(ctx)
-// 			if err != nil {
-// 				t.Fatalf("unexpected error initializing jetstream: %v", err)
-// 			}
-// 			err = jetStream.RecreateStream(ctx)
-// 			if err != nil {
-// 				t.Fatalf("unexpected error recreating jetstream: %v", err)
-// 			}
-// 			defer jetStream.Close()
-// 			err := testPublish(ctx, tt.tree)
-// 			if err != nil {
-// 				t.Fatalf("unexpected error on emit: %v", err)
-// 			}
-// 			var cancel context.CancelFunc
-//
-// 			ctx, cancel = context.WithTimeout(ctx, time.Second)
-// 			defer cancel()
-//
-// 			transportFunc := func(d []assembler.Graph, _ []*parser_common.IdentifierStrings) error {
-// 				if len(d) != len(tt.want) {
-// 					t.Errorf("ParseDocumentTree() = %v, want %v", d, tt.want)
-// 				}
-// 				for i := range d {
-// 					compare(t, d[i].Edges, tt.want[i].Edges, d[i].Nodes, tt.want[i].Nodes)
-// 				}
-// 				return nil
-// 			}
-//
-// 			err = Subscribe(ctx, transportFunc)
```

Comment:
We should revive this test as well to test the functionality for the NATS Subscriber.

Reply:
I don't want to make this PR too large, so I will do that in a subsequent PR.